drivers/kubevirt/kubevirt.go

```diff
@@ -507,3 +515,330 @@ func AddPVCsToVirtualMachine(vm kubevirtv1.VirtualMachine, pvcs []*corev1.Persis
+		return err
+	}
+	log.Infof("VM migration created for VM [%s] in namespace [%s]", vmiName, vmiNamespace)
+
+	// wait for completion
+	var migr *kubevirtdy.VirtualMachineInstanceMigration
+
+	// get volumes from app context
+	volume, err := schedDriver.GetVolumes(virtualMachineCtx)
+	if err != nil {
+		return err
+	}
+
+	t := func() (interface{}, bool, error) {
+		migr, err = kubevirtdy.Instance().GetVirtualMachineInstanceMigration(ctx, vmiNamespace, migration.Name)
+		if err != nil {
+			return "", false, fmt.Errorf("failed to get migration for VM [%s] in namespace [%s]", vmiName, vmiNamespace)
+		}
+		if !migr.Completed {
+			return "", true, fmt.Errorf("waiting for migration to complete for VM [%s] in namespace [%s]", vmiName, vmiNamespace)
+		}
+
+		// wait until there is only one pod in the running state
+		testPod, err := getVMPod(virtualMachineCtx, volume[0])
+		if err != nil {
+			return "", true, err
+		}
+
+		//Get the node where the vm is scheduled after the migration
+		nodeNameAfterMigration := testPod.Spec.NodeName
+
+		if nodeName == nodeNameAfterMigration {
+			return "", false, fmt.Errorf("VM pod live migrated [%s] in namespace [%s] but is still on the same node [%s]", testPod.Name, testPod.Namespace, nodeName)
+		}
+		log.InfoD("VM pod live migrated to node: [%s]", nodeNameAfterMigration)
+		return "", false, nil
+	}
+	_, err = task.DoRetryWithTimeout(t, defaultVmMountCheckTimeout, defaultVmMountCheckRetryInterval)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func getVMPod(virtualMachineCtx *scheduler.Context, vol *volume.Volume) (*corev1.Pod, error) {
+	pods, err := core.Instance().GetPodsUsingPV(vol.ID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pods for volume %s of context %s: %w", vol.ID, virtualMachineCtx.App.Key, err)
+	}
+
+	var found corev1.Pod
+	for _, pod := range pods {
+		if pod.Labels["kubevirt.io"] == "virt-launcher" && pod.Status.Phase == corev1.PodRunning {
+			if found.Name != "" {
+				// there should be only one VM pod in the running state (otherwise live migration is in progress)
+				return nil, fmt.Errorf("more than 1 KubeVirt pods (%s, %s) are in running state for volume %s",
+					found.Name, pod.Name, vol.ID)
+			}
+			found = pod
+		}
+	}
+	if found.Name == "" {
+		return nil, fmt.Errorf("failed to find a running pod for volume %s", vol.ID)
+	}
+	return &found, nil
+}
+
+func IsVMBindMounted(virtualMachineCtx *scheduler.Context, driver scheduler.Driver, volumeDriver volume.Driver, initialCheck bool) (bool, error) {
+	vols, err := driver.GetVolumes(virtualMachineCtx)
+	if err != nil {
+		return false, err
+	}
+	if len(vols) == 0 {
+		return false, fmt.Errorf("no volumes found for VM [%s] in namespace [%s]", virtualMachineCtx.App.Key, virtualMachineCtx.App.NameSpace)
+	}
+
+	// Get the node where pod is scheduled
+	vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{virtualMachineCtx})
+	if err != nil {
+		return false, err
+	}
+	if len(vms) != 1 {
```

Comment:
Add a comment for now that it will work only for 1 VM per spec/context. Ok to merge as of now but pls create a future ticket

Reply:
Done. Created a ticker https://purestorage.atlassian.net/browse/PTX-23167